src/containerapp/azext_containerapp/_up_utils.py

```diff
@@ -521,7 +527,7 @@ def build_container_from_source_with_cloud_build_service(self, source, build_env
 
         run_full_id = uuid.uuid4().hex
         logs_file_path = os.path.join(tempfile.gettempdir(), f"{'build{}'.format(run_full_id)[:12]}.txt")
-        logs_file = open(logs_file_path, "w")
+        logs_file = open(logs_file_path, "w", encoding="utf-8")
```

Comment:
Is the `encoding` part coming from this other PR? https://github.com/Azure/azure-cli-extensions/pull/7281/files
If yes, should we remove it from here (or take the rest of the changes so that we can close the other PR)?

Reply:
I left a comment for this PR https://github.com/Azure/azure-cli-extensions/pull/7281/files early this Week, about if there is any test cover this issue.
Do we have test case to cover this issue when code relies on UTF-8 logs?

Reply:
Merged these [changes](https://github.com/Azure/azure-cli-extensions/pull/7281/files) into this PR.